internal/version/version.go

```diff
@@ -15,19 +15,30 @@ import (
 //   - 0.6.0+release
 //   - 0.6.1
 //   - 0.6.2-alpha0+go1.21.nocgo
-const kwilVersion = "0.9.0-pre"
+const kwilVersion = "0." + MinorVersionV9 + ".0-pre"
+
+// MinorVersion is the minor version of the major version 0.
+type MinorVersion string
+
+// we start with 9 because we only started tracking this in v0.9.0
+const (
+	// MinorVersionV0 is the first minor version of the major version 0.
+	MinorVersionV9 MinorVersion = "9"
```

Comment:
The rest of this PR makes great sense, but I don't quite get this part with "first minor version of the major version 0" because I don't see how it evolves or how it's relevant in the context of this package.  This purpose of this `version` package would be better reflected by:

```
const (
	Major = "0"
	Minor = "9"
	Patch = "1"

	// kwilVersionDefault is the default KwilVersion string used when not set by
	// the linker at build time.
	kwilVersionDefault = Major + "." + Minor + "." + Patch + "-pre"
)
```

But for the purposes of the migration struct doesn't this feel like we want a "Migration Protocol Version", an int in the migration package, like there are block, p2p, and app protocol versions?

Reply:
Sure that makes sense

Reply:
Like if we didn't change the shape of `MigrationMetadata` between two major release (hopefully it eventually stabilizes), the integer version should stay the same not bump for no reason?

Reply:
or... "migration metadata version" if that's really the goal of allowing the receiver to interpret it appropriately.  So I guess migration *protocol* might suggest a much broader change than just the metadata structs.

Reply:
I just reverted all of the changes made in the version package. Figured we can do that later, not really in scope here

Reply:
> Like if we didn't change the shape of MigrationMetadata between two major release (hopefully it eventually stabilizes), the integer version should stay the same not bump for no reason

Yeah, correct.

Reply:
At this point, the versioning is only needed for stability of the `Snapshot` struct, since everything else is sent structured.